cmd/kots/cli/ingress.go

```diff
@@ -52,37 +51,46 @@ func IngressInstallCmd() *cobra.Command {
 				return err
 			}
 
-			ingressConfig := ingresstypes.Config{}
+			ingressConfig := kotsv1beta1.Ingress{}
 			if ingressConfigPath := v.GetString("ingress-config"); ingressConfigPath != "" {
 				content, err := ioutil.ReadFile(ingressConfigPath)
 				if err != nil {
-					return errors.Wrap(err, "failed to read ingress config file")
+					return errors.Wrap(err, "failed to read ingress service config file")
 				}
-				if err := ghodssyaml.Unmarshal(content, &ingressConfig); err != nil {
-					return errors.Wrap(err, "failed to unmarshal ingress config yaml")
+
+				s, err := ingress.DecodeSpec(content)
```

Comment:
i think this should be in `kotsutil` package for kotskinds for consistency. like this https://github.com/replicatedhq/kots/blob/656b706fb2e4e9b4b432b60af945d0e432b0c19f/pkg/kotsutil/kots.go#L476